### DISCUSSION TITLE
plantuml preview in markdown editor does not respect server settings

### DISCUSSION THREAD

**Discussion by ProgramFan** [requarks/wiki#3583] [Error / Bug Report]
**Describe the bug**
When the plantuml server in admin:rendering:markdown-html:plantuml is set to a custom server, the rendered page uses that server. But the plantuml preview in the markdown editor still uses https://plantuml.requarks.io. This causes the editor to not be able to display plantuml preview if works offline.

**To Reproduce**
1. Create a docker installation of wiki.js with plantuml server.
2. Configure the wiki to use http://localhost:8080 for plantuml server.
3. Disconnect the host from the Internet.
4. Create a plantuml diagram in the markdown editor.
4. See the preview displays "uml_diagram" instead of the preview.

**Host Info (please complete the following information):**
 - OS: Docker
 - Wiki.js version: 2.4.107
 - Database engine: sqlite 3.11

**Additional context**
Use developer tools of firefox on the preview shows that the url of the plantuml diagram points to "https://plantuml.requarks.io" while the url of the diagram in the rendered page points to "http://localhost:8080". The source code of plantuml.js of the markdown editor shows that it does accept a server configuration. So it might be a regession.

**Comment by NGPixel** [comment]
The client renderer is using a fixed configuration and won't reflect the server renderer. This will be addressed in a future update to have the client use the exact same configuration as the server.

**Comment by ProgramFan** [comment]
Thanks for your reply. Does this mean that there is a way to configure the client renderer? How can I configure it?

**Comment by NGPixel** [comment]
No, it's not configurable.

**Comment by AGrzes** [comment]
Is a PR on this welcome or is somebody else working on this enhancement?

**Comment by NGPixel** [comment]
@AGrzes As long as the PR addresses the full rendering pipeline, and not just PlantUML. If you're interested on working on this enhancement, open a draft PR first to discuss the implementation, before writing any code.

**Comment by AGrzes** [comment]
GtiHub is not allowing me to open draft PR without any change in code so I wlll put my proposed changes here.

1. Reproduce code fetching renders configuration from `client/components/admin/admin-rendering.vue` in `client/components/editor.vue`. 
2. Pass renderers configuration as a property to editor component
3. In `client/components/editor/editor-markdown.vue` Wrap `md` configuration in a function that accepts  renderers configuration
4. Create `md` as computed property of markdown editor component that will call the previously defined function to initialize markdown engine based on renderers config property
5. Use `md` computed property instead of `md` module variable.
6. Update `md` initialization to use renderers configuration
7. Update tests

Is this plan OK?

**Comment by AGrzes** [comment]
Hi - from my point of view the PR #2219 is ready for review - please let me know if anything need to be improved.

**Comment by lepeli** [comment]
Is there any update on this ?

It could be great to be able to change the editor's plantuml instance when working on sensitive data

**Comment by rsbueno** [comment]
The error still occurs in 2.5.297 version (docker). I've tried to set up a plant uml server to render a diagram using import directive (included in my plant uml server), in view mode, the diagram is showed without problems, but when I edit the page, importing error occurs. Inspecting network chrome tab, in edit-view the plant uml server used is https://plantuml.requarks.io/svg, not the configured.

**Comment by thickconfusion** [comment]
Still an issue in 2.5.301. This kind of defeats the purpose of self/internal hosting for those requests and their to be sent out to the internet.
